app/src/main/java/eu/kanade/tachiyomi/ui/manga/MangaDetailsPresenter.kt

```diff
@@ -230,8 +230,13 @@ class MangaDetailsPresenter(
         if (isLockedFromSearch) {
             return chapterList
         }
+        var observable = chapterList
         getScrollType(chapterList)
-        return chapterSort.getChaptersSorted(chapterList)
+        manga.filtered_scanlators?.let { filteredScanlatorString ->
+            val filteredScanlators = ChapterUtil.getScanlators(filteredScanlatorString)
+            observable = observable.filter { ChapterUtil.getScanlators(it.scanlator).any { group -> filteredScanlators.contains(group) } }
+        }
+        return chapterSort.getChaptersSorted(observable)
```

Comment:
This could instead be something like
```kotlin
        getScrollType(chapterList)
        val filteredChapterList = manga.filtered_scanlators?.let { filteredScanlatorString ->
            val filteredScanlators = ChapterUtil.getScanlators(filteredScanlatorString)
            observable = observable.filter { ChapterUtil.getScanlators(it.scanlator).any { group -> filteredScanlators.contains(group) } }
        } ?: chapterList
        return chapterSort.getChaptersSorted(filteredChapterList)
```

Reply:
I get type mismatch when trying to return `filteredChapterList`

Reply:
My b, it should be
```kotlin
        getScrollType(chapterList)
        val filteredChapterList = manga.filtered_scanlators?.let { filteredScanlatorString ->
            val filteredScanlators = ChapterUtil.getScanlators(filteredScanlatorString)
            return chapterList.filter { ChapterUtil.getScanlators(it.scanlator).any { group -> filteredScanlators.contains(group) } }
        } ?: chapterList
        return chapterSort.getChaptersSorted(filteredChapterList)
```